server/auth/adfs.ts

```diff
@@ -0,0 +1,98 @@
+import passport from "passport";
+import { IProfile, OIDCStrategy, VerifyCallback } from "passport-azure-ad";
+import express, { NextFunction, Request, Response } from "express";
+
+import "express-session";
+declare module "express-session" {
+  interface SessionData {
+    user: string;
+    passport: any;
+  }
+}
+
+const router = express.Router();
+
+passport.use(
+  new OIDCStrategy(
+    {
+      identityMetadata: `${process.env.ADFS_SERVER}/adfs/.well-known/openid-configuration`,
+      clientID: process.env.ADFS_CLIENT_ID ?? "",
+      responseType: "id_token",
+      responseMode: "form_post",
+      redirectUrl: process.env.REDIRECT_URI ?? "",
+      passReqToCallback: true,
+      useCookieInsteadOfSession: true,
+      cookieEncryptionKeys: [
+        {
+          key: process.env.COOKIE_ENCRYPTION_KEY ?? "",
+          iv: process.env.COOKIE_ENCRYPTION_IV ?? "",
```

Comment:
Can you add examples of these new environment variables to the .env.example?

Also, should we add an error message if they are detected to not exist? (feel free to make this its own ticket to deal with after ADFS implementation is done)

Reply:
I'll add examples yes fs. Also what should desired behavior be if a var isn't detected? Log an error and exit gracefully? or should the app continue running

Reply:
I think exit gracefully - most of these env variables are needed for the site to function securely, right?  We'd want the server admin to be very aware if something security-related is missing lol